library/core/src/slice/iter.rs

```diff
@@ -1613,10 +1608,7 @@ impl<'a, T> DoubleEndedIterator for Chunks<'a, T> {
             None
         } else {
             let start = (len - 1 - n) * self.chunk_size;
-            let end = match start.checked_add(self.chunk_size) {
-                Some(res) => cmp::min(self.v.len(), res),
-                None => self.v.len(),
-            };
+            let end = (start + self.chunk_size).min(self.v.len());
```

Comment:
This isn't correct for ZSTs if the array has length `usize::MAX`. `start + self.chunk_size` is equal to `len * self.chunk_size` and thus `self.v.len().div_ceil(self.chunk_size) * self.chunk_size`/`usize::MAX.div_ceil(self.chunk_size) * self.chunk_size`, which will overflow for a chunk size larger than one.